client/src/components/AuthModal.jsx

```diff
@@ -0,0 +1,282 @@
+    top: "50%",
+    transform: "translate(-50%, -50%)",
+    backgroundColor: "#f2f2f2",
+  },
+  textField: {
+    borderRadius: `4px 4px 4px 4px`,
+    outline: "none",
+    padding: "7px",
+    width: "250px",
+    backgroundColor: "white",
+  },
+  textFieldInput: {
+    width: "calc(100% - 24px)",
+  },
+  typography: {
+    textAlign: "center",
+    display: "block",
+  },
+  box: {
+    marginBottom: "20px",
+  },
+  h4: {
+    textAlign: "center",
+    margin: "20px 0 20px 0",
+  },
+  button: {
+    height: 56,
+    borderRadius: 28,
+    width: 142,
+    display: "block",
+    margin: "auto",
+    marginBottom: "20px",
+  },
+  footer: {
+    textAlign: "center",
+    padding: "20px 20px 20px 20px ",
+  },
+  hyperlink: {
+    display: "inline",
+    color: "#DF1B1B",
+    fontWeight: "bold",
+    cursor: "pointer",
+  },
+}));
+
+const AuthModal = ({ isAuthPage }) => {
+  const [email, setEmail] = useState("");
+  const [name, setName] = useState("");
+  const [password, setPassword] = useState("");
+  const [signedIn, setSignedIn] = useState(true);
+  const [isFirst, setIsFirst] = useState(true);
+
+  const history = useHistory();
+
+  useEffect(async () => {
+    setIsFirst(false);
+    let results = await fetch("/user/getCurrentUser");
+
+    results = await results.json();
+
+    if (results.user === undefined) {
+      setSignedIn(false);
+    } else {
+      setSignedIn(true);
+      history.push("/dashboard");
+    }
+  }, []);
+
+  const submitHandler = async (e) => {
+    e.preventDefault();
+
+    if (password.length < 7) {
+      alert("Password must be greater than 6");
+      return;
+    }
+
+    if (name.trim().length === 0 && pathname === "/signup") {
+      alert("Name must not be just spaces");
+      return;
+    }
+
+    try {
+      //email is handled by the TextField
+      if (pathname === "/signup") {
+        let result = await fetch("/user/register", {
+          method: "POST",
+          headers: { "Content-type": "application/json" },
+          body: JSON.stringify({
+            name,
+            email,
+            password,
+          }),
+        });
+
+        result = await result.json();
+
+        setSignedIn(
+          result.user !== undefined && result.user.id !== undefined
+            ? true
+            : false
+        );
+      } else {
+        let result = await fetch("/user/login", {
+          method: "POST",
+          headers: { "Content-type": "application/json" },
+          body: JSON.stringify({
+            email,
+            password,
+          }),
+        });
+
+        result = await result.json();
+
+        setSignedIn(
+          result.user !== undefined && result.user.id !== undefined
+            ? true
+            : false
+        );
+      }
+      history.push("/dashboard");
+    } catch (error) {
+      alert("Something went wrong");
+      console.error(error);
+    }
+  };
+
+  const classes = useStyles();
+  const { pathname } = window.location;
```

Comment:
There's a couple of things I want to point out about this line:

- When using React, it's a good idea to not reach out into the window or document at all. Let React handle that. If you want access to the pathname, you can use the `location` prop that is available to you if you navigated to this component using React Router, or you can alternatively use the [`useLocation`](https://reactrouter.com/web/api/Hooks) hook that React Router gives you if you didn't navigate to this component directly using a `<Route />`.
- You are declaring the pathname in line 155, but you're actually using it in 104 and 111. Even though this still works in the context of React (because it's "compiled" into regular JS), it's still good practice to declare these definitions at the start of the component.